CDP4ParameterSheetGenerator/Assemblers/OptionSheet/OptionSheetRowAssembler.cs

```diff
@@ -120,6 +120,29 @@ public void Assemble()
             foreach (var nestedElementExcelRow in nestedElementExcelRows)
             {
                 var deeprows = nestedElementExcelRow.GetContainedRowsDeep();
+
+                foreach (var itm in deeprows)
+                {
+                    var splitlevel = itm.ModelCode.Split('\\')[0].Split('.').Length;
+
+                    if (itm.Categories == null)
```

Comment:
The `Categories` of a Thing should play no role in determining its level.

Reply:
The categories is used in correlation with the node key in ModelCode. When ther are no categories the ModeCode holds the key in the first segment same as the parent. Otherwise the key is the actual key for instance Loft.Satellite.10 layers while in case of diameter the key is still Loft.Satellite.10 layers and in the second segment stores the name as Loft.Satelitte.10 layers\d\default while "10 layers"is Loft.Satelitte.10 layers\default